tests/virt_autotest/virsh_internal_snapshot.pm

```diff
@@ -33,28 +33,30 @@ sub run_test {
     return unless check_var("REGRESSION", "qemu-hypervisor") || check_var("SYSTEM_ROLE", "kvm");
 
     foreach my $guest (keys %xen::guests) {
-        record_info "virsh-snapshot", "Creating Internal Snapshot";
-        assert_script_run "virsh snapshot-create-as $guest --name internal-snapshot-$guest-01";
-        assert_script_run "virsh snapshot-current $guest | grep internal-snapshot-$guest-01";
-        assert_script_run "virsh snapshot-list $guest | grep internal-snapshot-$guest-01";
-        assert_script_run "virsh snapshot-create-as $guest --name internal-snapshot-$guest-02";
-        assert_script_run "virsh snapshot-list $guest | grep internal-snapshot-$guest-02";
+        if (check_guest_disk_type($guest)) {
```

Comment:
next if not check_guest_disk_type($guest) will need much less code change

Reply:
Had updated code here.
Thank you so much for your great help.